packages/react/src/components/MultiSelect/FilterableMultiSelect.tsx

```diff
@@ -728,7 +737,7 @@ const FilterableMultiSelect = React.forwardRef(function FilterableMultiSelect<
       placeholder,
       preventKeyAction: isOpen,
 
-      onClick: () => handleMenuChange(true),
+      onClick: () => !readOnly && handleMenuChange(true),
```

Comment:
We are checking `readOnly` condition here and in `handleMenuChange` function both. I don't think its required in both places.